pyfar/signals/signals.py

```diff
@@ -0,0 +1,568 @@
+
+    Parameters
+    ----------
+    n_pulse : int
+        The length of the pulses in samples
+    n_pause : int
+        The length of the pauses between the pulses in samples.
+    n_fade : int, optional
+        The length of the squared sine/cosine fade-in and fade outs in samples.
+        The default is 90, which equals approximately 2 ms at sampling rates of
+        44.1 and 48 kHz.
+    repetitions : int, optional
+    rms : double, array like, optional
+        The RMS amplitude of the white noise signal. The default is 1.
+    color: string, optional
+        The noise color, which can be 'pink' or 'white'. The default is 'pink'.
+    frozen : boolean, optional
+        If True, all noise pulses are identical. If False each noise pulse is
+        a separate stochastic process. The default is True.
+    sampling_rate : int, optional
+        The sampling rate in Hz. The default is 44100.
+    seed : int, None
+        The seed for the random generator. Pass a seed to obtain identical
+        results for multiple calls of white noise. The default is None, which
+        will yield different results with every call.
+
+    Returns
+    -------
+    signal : Signal
+        The noise as a Signal object. The Signal is in the time domain and
+        has the 'rms' FFT normalization (see pyfar.fft.normalization).
+    """
+
+    if n_pulse < 2 * n_fade:
+        raise ValueError(
+            "n_fade too large. It must be smaller than n_pulse/2.")
+
+    # get the noise sample
+    n_pulse = int(n_pulse)
+    repetitions = int(repetitions)
+    n_samples = n_pulse if frozen else n_pulse * repetitions
+
+    if color == "pink":
+        noise = pink_noise(n_samples, rms, sampling_rate, seed).time
+    elif color == "white":
+        noise = white_noise(n_samples, rms, sampling_rate, seed).time
+    else:
+        raise ValueError(f"color is {color} but must be 'pink' or 'white'.")
+
+    noise = np.tile(noise, (repetitions, 1)) if frozen else \
+        noise.reshape((repetitions, n_pulse))
+
+    # fade the noise
+    if n_fade > 0:
+        n_fade = int(n_fade)
+        fade = np.sin(np.linspace(0, np.pi/2, n_fade))**2
+        noise[..., 0:n_fade] *= fade
+        noise[..., -n_fade:] *= fade[::-1]
+
+    # add the pause
+    noise = np.concatenate((noise, np.zeros((repetitions, int(n_pause)))), -1)
+
+    # reshape to single channel signal and discard final pause
+    noise = noise.reshape((1, -1))[..., :-int(n_pause)]
+
+    # save to Signal
+    frozen_str = "frozen" if frozen else ""
+    comment = (f"{frozen_str} {color} pulsed noise signal (rms = {rms}, "
+               f"{repetitions} repetitions, {n_pulse} samples pulse duration, "
+               f"{n_pause} samples pauses, and {n_fade} samples fades.")
+
+    signal = Signal(noise, sampling_rate, fft_norm="rms", comment=comment)
+
+    return signal
+
+
+def linear_sweep(n_samples, frequency_range, n_fade=90, amplitude=1,
+                 sampling_rate=44100):
```

Comment:
1. I would rename to `n_fade_out`. Helps avoiding confusion with fade-in windows.
2. Should we include a phase angle as well?